text/0014-improve-locking-mechanism-for-parachains.md

```diff
@@ -0,0 +1,105 @@
+# RFC-0014: Improve locking mechanism for parachains
+
+|                 |                                          |
+| --------------- | ---------------------------------------- |
+| **Start Date**  | July 25, 2023                            |
+| **Description** | Improve locking mechanism for parachains |
+| **Authors**     | Bryan Chen                               |
+
+## Summary
+
+The manager of a parachain have the permission to manage the parachain when the parachain is unlocked. Parachain is by default locked when onboarded on a slot. This requires the parachain wasm/genesis must be valid otherwise a root track governance action on relaychain is required to update the parachain.
+
+This RFC proposes a mechanism to allow parachain manager self-service the parachain without root track governance action.
+
+## Motivation
+
+The current reliance on root track governance actions for managing parachains can be time-consuming and burdensome. This RFC aims to address this technical difficulty by allowing parachain managers to take self-service actions, rather than relying on general public voting.
+
+The key scenarios this RFC seeks to improve are:
+
+1. Rescue a parachain with invalid wasm/genesis.
+
+While we have various resources and templates to build a new parachain, it is still not a trivial task. It is very easy to make a mistake and resulting an invalid wasm/genesis. With lack of tools to help detect those issues[^1], it is very likely that the issues are only discovered after the parachain is onboarded on a slot. In this case, the parachain is locked and the parachain team has to go through a lengthy governance process to rescue the parachain.
+
+2. Perform lease renewal for an existing parachain.
+
+One way to perform lease renewal for a parachain is by perform a least swap with another parachain with longer lease. This requires the other parachain must be operational and able to perform XCM transact call into relaychain to dispatch the swap call. Combined with the overhead of setting up a new parachain, this is an time consuming and expensive process. Ideally, the parachain manager should be able to perform the lease swap call without having a running parachain[^2].
+
```

Comment:
Could you add some hard requirements in MUST/SHOULD terms?

Reply:
added